src/builtins.ts

```diff
@@ -2188,48 +2191,49 @@ export function compileCall(
         WrapMode.NONE
       );
       let offset = operands.length == 2 ? evaluateImmediateOffset(compiler, operands[1]) : 0; // reports
-      if (offset < 0) { // reported in evaluateImmediateOffset
-        return module.createUnreachable();
-      }
-      compiler.currentType = typeArguments[0];
+      if (offset < 0) return module.createUnreachable();
+      compiler.currentType = loadType;
       return module.createAtomicLoad(
-        typeArguments[0].byteSize,
+        loadType.byteSize,
         arg0,
-        typeArguments[0].is(TypeFlags.INTEGER) &&
+        loadType.is(TypeFlags.INTEGER) &&
         contextualType.is(TypeFlags.INTEGER) &&
-        contextualType.size > typeArguments[0].size
+        contextualType.size > loadType.size
           ? (compiler.currentType = contextualType).toNativeType()
-          : (compiler.currentType = typeArguments[0]).toNativeType(),
+          : (compiler.currentType = loadType).toNativeType(),
```

Comment:
Hmm, it's little bit unusual style. May be better:
```ts
compiler.currentType = contextualType.size > loadType.size 
  ? contextualType.toNativeType()
  : loadType.toNativeType()
```

Reply:
Hmm, that would assign a `NativeType` (enum/i32) to a `Type` (object).

Reply:
Ahh, miss this a complex expression)